pkg/apis/core/v1alpha2/types.go

```diff
@@ -155,3 +155,27 @@ type ClusterGroupList struct {
 
 	Items []ClusterGroup `json:"items,omitempty"`
 }
+
+// AppliedTo selects the entities to which a policy is applied.
+type AppliedTo struct {
+	// Selectors is the set of EntitySelectors that select entities.
+	// +optional
+	Selectors []EntitySelector `json:"selectors,omitempty"`
+	// Groups is the set of ClusterGroup names.
+	// +optional
+	Groups []string `json:"groups,omitempty"`
+}
+
+// EntitySelector includes the selectors that select the matched entities.
+type EntitySelector struct {
+	// Select Pods matched by this selector. If set with NamespaceSelector,
+	// Pods are matched from Namespaces matched by the NamespaceSelector;
+	// otherwise, Pods are matched from all Namespaces.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+	// Select all Pods from Namespaces matched by this selector, as
+	// workloads in To/From fields. If set with PodSelector,
+	// Pods are matched from Namespaces matched by the NamespaceSelector.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
```

Comment:
Not related to this PR, I'm wondering if we could add a Namespace field to the EntitySelector which means selecting Pods in this namespace, just like how NetworkPolicyPeer in K8s NetworkPolicy works. Is this easier to use compared with "antrea.io/metadata.name" or "kubernetes.io/metadata.name" label?

Reply:
I think Abishek and Yang proposed the label way, because upstream NetworkPolicy decide to go that direction. @Dyanngg: please comment.

Reply:
Thanks for bringing it up. In the upstream proposal, we're replacing the `NamespaceSelector` field with a new `Namespaces` field, which look like the following
```
type Namespaces struct {
	Self       bool
	Selector   *metav1.LabelSelector
	Except     []*metav1.LabelSelector
}
```
`Namespaces.Selector` will essentially replace `NamespaceSelector`. And user can easily write a Namespace isolation policy by setting `Namespaces.Self=true`. We are still on the fence about whether the `Except` field is necessary so I wouldn't put that in to start with (although there are quite some usecases which can be greatly simplified by `except`, like deny all traffic except from kube-system). 
Link to the KEP: https://github.com/kubernetes/enhancements/pull/2522